tests/approaches/test_residual_approach.py

```diff
@@ -0,0 +1,113 @@
+"""Tests for the ResidualApproach (SB3 TD3/DDPG backends) on Pendulum-v1."""
+
+# pylint: disable=protected-access
+
+import contextlib
+import io
+import sys
+from typing import Any, Literal, TextIO, cast
+
+import gymnasium as gym
+import numpy as np
+import pytest
+
+from programmatic_policy_learning.approaches.pendulum_stupid_approach import (
+    PendulumStupidAlgorithm,
+)
+from programmatic_policy_learning.approaches.residual_approach import ResidualApproach
+
+ENV_ID = "Pendulum-v1"
+SEED = 42
+TRAIN_STEPS = 10_000
+EVAL_EPISODES = 5
+BACKENDS: list[Literal["sb3-td3", "sb3-ddpg"]] = ["sb3-td3", "sb3-ddpg"]
+
+
+def build_env() -> gym.Env:
+    """Construct a fresh Gymnasium environment."""
+    return gym.make(ENV_ID)
+
+
+def eval_policy(
+    approach: Any, n_episodes: int = EVAL_EPISODES, seed: int = SEED
+) -> np.ndarray:
+    """Run n_episodes and return per-episode returns; suppress base prints so
+    output is clean."""
+    returns: list[float] = []
+    for ep in range(n_episodes):
+        env = build_env()
+        obs, _ = env.reset(seed=seed + ep)
+        terminated = truncated = False
+        ep_ret = 0.0
+        while not (terminated or truncated):
+            buf = io.StringIO()
+            with contextlib.redirect_stdout(buf):
+                approach._last_observation = np.asarray(
+                    obs, dtype=np.float32
+                )  # pylint: disable=protected-access
+                a = approach._get_action()  # pylint: disable=protected-access
+            obs, r, terminated, truncated, _ = env.step(a)
+            ep_ret += float(r)
+        cast(Any, env).close()  # mypy: env.close is untyped
+        returns.append(ep_ret)
+    return np.asarray(returns, dtype=np.float32)
+
+
+@pytest.mark.parametrize("backend", BACKENDS)
+def test_residual_vs_base_runs(backend: Literal["sb3-td3", "sb3-ddpg"]) -> None:
+    """Smoke test: residual approach runs and is not catastrophically worse than base."""
```

Comment:
this looks great, but I suspect that either this test or another one is slow. we will do larger scale experiments outside the unit tests -- we want to make sure the unit tests are as fast as possible. it probably won't be feasible to really test that RL works within unit tests, so just making sure things run is good